### PR TITLE
Add `pallet-teerex` tests to CI with all feature combinations

### LINKED ISSUE #2139: Parachain pallets tests not run properly in CI
I think there is a problem in running parachain's pallet unit tests. At least for me it gives false-positive results. 

In `parachain-unit-tests` GA job there is command:
```
cargo test --locked --release -p pallet-* --lib --features=skip-ias-check
```

And once it's run on current dev, everything looks fine, tests pass.

But once I try to run tests for specific pallet like

```
cargo test --locked --release -p pallet-teerex --lib --features=skip-ias-check
```

The code cannot be even compiled:

```
kziemianek@kziemianek:~/projects/litentry/litentry-parachain$ cargo test --locked --release -p pallet-teerex --lib --features=skip-ias-check
   Compiling pallet-teerex v0.9.0 (/home/kziemianek/projects/litentry/litentry-parachain/pallets/teerex)
error[E0308]: mismatched types
   --> pallets/teerex/src/tests/skip_ias_check_tests.rs:115:4
    |
113 |         assert_ok!(Teerex::update_scheduled_enclave(
    |                    -------------------------------- arguments to this function are incorrect
114 |             RuntimeOrigin::root(),
115 |             0u32,
    |             ^^^^ expected `u64`, found `u32`
    |
note: associated function defined here
   --> pallets/teerex/src/lib.rs:476:10
    |
476 |         pub fn update_scheduled_enclave(
    |                ^^^^^^^^^^^^^^^^^^^^^^^^
477 |             origin: OriginFor<T>,
478 |             #[pallet::compact] sidechain_block_number: SidechainBlockNumber,
    |                                --------------------------------------------
help: change the type of the numeric literal from `u32` to `u64`
    |
115 |             0u64,
    |              ~~~

For more information about this error, try `rustc --explain E0308`.
error: could not compile `pallet-teerex` due to previous error
```

### PR DESCRIPTION
**Pull request by Kailai-Wang**
### Context

fixes #2139 

A minor PR - that's the only compilation error that I found. The fact that the tests fail for some features is known.

It's also known behavior that when using wildcards(`pallet-*`) the features will be consolidated, so it might behave differently from using concrete `pallet-teerex`.

### CODE REVIEW

**Comment by kziemianek** [comment]
Just wondering if there is a way to catch this kind of errors in CI and whether it's worth doing so.

**Review by kziemianek — APPROVED** [review]

**Comment by Kailai-Wang** [comment]
I added the test commands to CI, with all feature guards combinations - need a second review :)

File: .github/workflows/ci.yml
Comment (kziemianek) [inline .github/workflows/ci.yml]: Looks like a hell to maintain :(

**Review by kziemianek — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (kziemianek) [reply .github/workflows/ci.yml]: https://github.com/taiki-e/cargo-hack#--feature-powerset 🤔

Comment (kziemianek) [reply .github/workflows/ci.yml]: Not sure if the command is ok, but I managed to get those commands running from single `cargo hack` command

```
kziemianek@kziemianek ~/p/l/litentry-parachain (p-146-parachain-pallets-tests-not-run-properly-in-ci) [1]> cargo hack test --feature-powerset --features=default --locked --release -p pallet-teerex --lib
info: running `cargo test --locked --release --lib --no-default-features --features default` on pallet-teerex (1/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks` on pallet-teerex (2/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check` on pallet-teerex (3/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check` on pallet-teerex (4/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-scheduled-enclave-check` on pallet-teerex (5/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-scheduled-enclave-check` on pallet-teerex (6/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,skip-scheduled-enclave-check` on pallet-teerex (7/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,skip-scheduled-enclave-check` on pallet-teerex (8/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,std` on pallet-teerex (9/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,std` on pallet-teerex (10/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,std` on pallet-teerex (11/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,std` on pallet-teerex (12/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-scheduled-enclave-check,std` on pallet-teerex (13/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-scheduled-enclave-check,std` on pallet-teerex (14/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,skip-scheduled-enclave-check,std` on pallet-teerex (15/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,skip-scheduled-enclave-check,std` on pallet-teerex (16/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,try-runtime` on pallet-teerex (17/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,try-runtime` on pallet-teerex (18/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,try-runtime` on pallet-teerex (19/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,try-runtime` on pallet-teerex (20/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-scheduled-enclave-check,try-runtime` on pallet-teerex (21/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-scheduled-enclave-check,try-runtime` on pallet-teerex (22/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,skip-scheduled-enclave-check,try-runtime` on pallet-teerex (23/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,skip-scheduled-enclave-check,try-runtime` on pallet-teerex (24/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,std,try-runtime` on pallet-teerex (25/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,std,try-runtime` on pallet-teerex (26/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,std,try-runtime` on pallet-teerex (27/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,std,try-runtime` on pallet-teerex (28/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-scheduled-enclave-check,std,try-runtime` on pallet-teerex (29/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-scheduled-enclave-check,std,try-runtime` on pallet-teerex (30/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,skip-scheduled-enclave-check,std,try-runtime` on pallet-teerex (31/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,skip-scheduled-enclave-check,std,try-runtime` on pallet-teerex (32/32)
```

Comment (kziemianek) [reply .github/workflows/ci.yml]: I don't think we want to run tests this way, just treat it as a curiosity. 🙃

Comment (Kailai-Wang) [reply .github/workflows/ci.yml]: It looks great - although all 32 combinations might be a bit overkill 😂 

Let me integrate it into the CI

**Pull request merged**
